### PR TITLE
Added functional tests for 2.0

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

File: test/EmptyApp20.FunctionalTests/FunctionalTest/RequestTelemetryEmptyAppTests.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
+﻿namespace EmptyApp20.FunctionalTests.FunctionalTest
+{
+    using System.Linq;
+    using System.Net.Http;
+    using FunctionalTestUtils;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Xunit;
```
Comment (Dmitry-Matveev) [inline test/EmptyApp20.FunctionalTests/FunctionalTest/RequestTelemetryEmptyAppTests.cs]: It looks like Sergey was removing XUnit with [this PR](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/608/files), so we might want to use MSTEST here instead as well.

File: test/FunctionalTestUtils20/FunctionalTestUtils20.csproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <VersionPrefix>1.0.2</VersionPrefix>
+    <TargetFrameworks>net461;netstandard2.0</TargetFrameworks>
+    <DelaySign>true</DelaySign>
+    <PreserveCompilationContext>true</PreserveCompilationContext>
+    <AssemblyName>FunctionalTestUtils20</AssemblyName>
+    <AssemblyOriginatorKeyFile>../../keys/35MSSharedLib1024.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
+    <PackageId>FunctionalTestUtils</PackageId>
+    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
```
Comment (Dmitry-Matveev) [inline test/FunctionalTestUtils20/FunctionalTestUtils20.csproj]: Should it be 1.6.1 or something like 2.0?

Comment (cijothomas) [reply test/FunctionalTestUtils20/FunctionalTestUtils20.csproj]: Good catch! 2.0.

File: test/EmptyApp20.FunctionalTests/FunctionalTest/RequestTelemetryEmptyAppTests.cs (hunk 1 above)
Comment (cijothomas) [reply test/EmptyApp20.FunctionalTests/FunctionalTest/RequestTelemetryEmptyAppTests.cs]: The issue faced in other repo is not applicable in aspnet core repro as we don't target anything below net451. 
We can replace it in future if need arises, but as of now i dont see strong reason to remove it. It'll require some work as the output logging is based on xunit now.

**Pull request merged**